internal/fs/vss_windows.go

```diff
@@ -0,0 +1,986 @@
+			uintptr(unsafe.Pointer(&nondeletedSnapshotID)))
+	}
+
+	err := newVssErrorIfResultNotOK("DeleteSnapshots() failed", HRESULT(result))
+	return deletedSnapshots, nondeletedSnapshotID, err
+}
+
+// GetSnapshotProperties calls the equivalent VSS api.
+func (vss *IVssBackupComponents) GetSnapshotProperties(snapshotID ole.GUID,
+	properties *VssSnapshotProperties) error {
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		ptrToID := uintptr(unsafe.Pointer(&snapshotID))
+
+		result, _, _ = syscall.Syscall6(vss.getVTable().getSnapshotProperties, 6,
+			uintptr(unsafe.Pointer(vss)), uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 4)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 8)))),
+			uintptr(*(*uintptr)(unsafe.Pointer(uintptr(ptrToID + 12)))),
+			uintptr(unsafe.Pointer(properties)))
+	} else {
+		result, _, _ = syscall.Syscall(vss.getVTable().getSnapshotProperties, 3,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotID)),
+			uintptr(unsafe.Pointer(properties)))
+	}
+
+	return newVssErrorIfResultNotOK("GetSnapshotProperties() failed", HRESULT(result))
+}
+
+// vssFreeSnapshotProperties calls the equivalent VSS api.
+func vssFreeSnapshotProperties(properties *VssSnapshotProperties) error {
+	proc, err := findVssProc("VssFreeSnapshotProperties")
+	if err != nil {
+		return err
+	}
+
+	proc.Call(uintptr(unsafe.Pointer(properties)))
+	return nil
+}
+
+// BackupComplete calls the equivalent VSS api.
+func (vss *IVssBackupComponents) BackupComplete() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().backupComplete, 2, uintptr(unsafe.Pointer(vss)),
+		uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := newVssErrorIfResultNotOK("BackupComplete() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// VssSnapshotProperties defines the properties of a VSS snapshot as part of the VSS api.
+type VssSnapshotProperties struct {
+	snapshotID           ole.GUID
+	snapshotSetID        ole.GUID
+	snapshotsCount       uint32
+	snapshotDeviceObject *uint16
+	originalVolumeName   *uint16
+	originatingMachine   *uint16
+	serviceMachine       *uint16
+	exposedName          *uint16
+	exposedPath          *uint16
+	providerId           ole.GUID
+	snapshotAttributes   uint32
+	creationTimestamp    uint64
+	status               uint
+}
+
+// GetSnapshotDeviceObject returns root path to access the snapshot files
+// and folders.
+func (p *VssSnapshotProperties) GetSnapshotDeviceObject() string {
+	return ole.UTF16PtrToString(p.snapshotDeviceObject)
+}
+
+// UIID_IVSS_ASYNC defines to GUID of IVSSAsync.
+var UIID_IVSS_ASYNC = &ole.GUID{
+	0x507C37B4, 0xCF5B, 0x4e95,
+	[8]byte{0xb0, 0xaf, 0x14, 0xeb, 0x97, 0x67, 0x46, 0x7e},
+}
+
+// IVSSAsync VSS api interface.
+type IVSSAsync struct {
+	ole.IUnknown
+}
+
+// IVSSAsync is the vtable for IVSSAsync.
+type IVSSAsyncVTable struct {
```

Comment:
// IVSSAsync**VTable** is the vtable for IVSSAsync.

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/97322db98cdc31a9aba2c358e59809418cdf68be